lib/src/models/multireddit.dart

```diff
@@ -3,13 +3,398 @@
+  String get infoPath => _infoPath ?? '/';
+  String _infoPath;
+
+  /// The visibility of this multireddit.
+  ///
+  /// Refer to [Visibility]'s Enum definition for more information.
+  /// If this information is not provided, will return null.
+  Visibility get visibility => Visibility.values.firstWhere(
+      (e) => e.toString() == ('Visibility.' + data['visibility']),
+      orElse: () => null);
+
+  /// The weightingScheme of this multireddit.
+  ///
+  /// Refer to [WeightingScheme]'s Enum definition for more information.
+  /// If this information is not provided, will return null.
+  WeightingScheme get weightingScheme => WeightingScheme.values.firstWhere(
+      (e) => e.toString() == ('WeightingScheme.' + data['weighting_scheme']),
+      orElse: () => null);
+
+  /// Does the currently authenticated [User] have the privilege to edit this
+  /// multireddit.
+  bool get canEdit => data['can_edit'];
+
+  /// Does this multireddit require visitors to be over the age of 18.
+  bool get over18 => data['over_18'];
+
   Multireddit.parse(Reddit reddit, Map data) : super(reddit) {
+    if (!data['data'].containsKey('name')) {
+      throw new DRAWUnimplementedError();
+    }
     setData(this, data['data']);
+    _author = new RedditorRef.name(
+        reddit, data['data']['path']?.split('/')[_redditorNameInPathIndex]);
+    _infoPath = _generateInfoPath(data['data']['name'], _author.displayName);
+  }
+
+  // Returns valid info_path for multireddit with name `name`.
+  static String _generateInfoPath(String name, String user) =>
+      apiPath['multireddit']
+          .replaceAll(_multiredditRegExp, name)
+          .replaceAll(_userRegExp, user);
+
+  // Returns `str` in camel case.
+  static String _convertToCamelCase(String str) {
+    if (str == null) {
+      return null;
+    }
+    final splitStrList = str.split(new RegExp(r'(\s|_)+'));
+    final strItr = splitStrList.iterator;
+    final strBuffer = new StringBuffer(strItr.toString());
+    while (strItr.moveNext()) {
+      strBuffer.write(strItr.toString().substring(0, 1).toUpperCase());
+      strBuffer.write(strItr.toString().substring(1));
+    }
+    return strBuffer.toString();
+  }
+
+  /// Returns a slug version of the `title`.
+  static String _sluggify(String title) {
+    if (title == null) {
+      return null;
+    }
+    final RegExp _invalidRegExp = new RegExp(r'(\s|\W|_)+');
+    var titleScoped = title.replaceAll(_invalidRegExp, '_').trim();
+    if (titleScoped.length > 21) {
+      titleScoped = titleScoped.substring(21);
+      final lastWord = titleScoped.lastIndexOf('_');
+      //TODO(ckartik): Test this well.
+      if (lastWord > 0) {
+        titleScoped = titleScoped.substring(lastWord);
+      }
+    }
+    return titleScoped ?? '_';
+  }
+
+  /// Add a [Subreddit] to this [Multireddit].
+  ///
+  /// `subreddit` is the name of the [Subreddit] to be added to this [Multireddit].
+  Future add(/* String, Subreddit */ subreddit) async {
```

Comment:
Check for `String` and `Subreddit` and throw `DRAWArgumentError` if `subreddit` is neither.

Reply:
Already doing this in the subredditHelper.

Reply:
I will add it to the documentation that this procedure throws, let me know if you would like me to refactor the code to be more readable!